Program.cs

```diff
@@ -0,0 +1,410 @@
+                MenuPrincipal();
+                ProcesoPrincipal();
+            }
+
+            void RegistryInvoice()
+            {
+                Console.WriteLine("Codigo Cliente: ");
+                if (!int.TryParse(Console.ReadLine(), out int idCustomer))
+                {
+                    return;
+                }
+
+                Customers objCustomers = customersList.SingleOrDefault(cliente => cliente.IdCustomer == idCustomer);               
+
+                Console.WriteLine($"Cliente Seleccionado: {objCustomers.Name} , Fecha de Nacimiento: {objCustomers.BirthDate}");
+
+                Console.WriteLine("Codigo Vendedor: ");
+                if (!int.TryParse(Console.ReadLine(), out int idSeller))
+                {
+                    return;
+                }
+
+                Seller objSeller = sellerList.SingleOrDefault(vendedor => vendedor.IdSeller == idSeller);
+
+                Console.WriteLine($"Vendedor Seleccionado: {objSeller.Name}");
+
+                //Obtenemos el ultimo ID
+                int maxIdInvoice = 1;
+
+                if (invoiceList.Count > 0) //Otro metodo
+                {
+                    maxIdInvoice = invoiceList.Max(c => c.IdInvoice) + 1;
+                }
+
+                //Se graba la cabecera
+                Invoice invoice = new Invoice();
+                invoice.IdInvoice = maxIdInvoice;
+                invoice.Customer = objCustomers;
+                invoice.Seller = objSeller;
+
+                //invoiceList.Add(invoice);
+
+                string optionSelected = "y";
+                int numberRegistry = 0;
+                decimal total = 0;
+
+                invoice.InvoiceDetails = new List<InvoiceDetail>();
+
+                while (optionSelected == "y")
+                {
+                    Console.WriteLine("Codigo Producto: ");
+                    if (!int.TryParse(Console.ReadLine(), out int idProduct))
+                    {
+                        return;
+                    }
+
+                    Product objProduct = productList.SingleOrDefault(producto => producto.IdProduct == idProduct);
+
+                    PrintData(objProduct);
+
+                    Console.WriteLine("Cantidad: ");
+                    if (!int.TryParse(Console.ReadLine(), out int quantity))
+                    {
+                        return;
+                    }
+
+                    InvoiceDetail invoiceDetail = new InvoiceDetail();
+                    invoiceDetail.IdInvoiceDetail = numberRegistry + 1;
+                    invoiceDetail.Product = objProduct;
+                    invoiceDetail.Price = objProduct.Price;
+                    invoiceDetail.Quantity = quantity;
+
+                    total = total + objProduct.Price*quantity;
+                    
+                    invoice.InvoiceDetails.Add(invoiceDetail);
+
+                    numberRegistry++;
+
+                    Console.WriteLine("Desea agregar detalle de la factura?: (y/n)");

```

Comment:
When You see a code repeated almost the same, as making a request and wait for an answer, maybe could be time to extract a method.